### DISCUSSION TITLE
How to use code generator with custom AWS AppSync scalars?

### DISCUSSION THREAD

**Discussion by xzilja** [dotansimha/graphql-code-generator#5283] [Q&A] [answered]
I am creating my graphql schema for AWS AppSync. They expose some custom scalars as per https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html#appsync-defined-scalars

I tried adding these in my `codegen.yml` as follows
```yml
schema: ../server/src/schema.graphql
generates:
  ./typings/api.ts:
    plugins:
      - typescript
    config:
      scalars:
        AWSDateTime: Date
```

However am still getting following error
```zsh
$ graphql-codegen
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate ./typings/api.ts
      ✖ Load GraphQL schemas
        → Failed to load schema
        Load GraphQL documents
        Generate


 Found 1 error

  ✖ ./typings/api.ts
    Failed to load schema from ../server/src/schema.graphql:

        Unknown type: "AWSDateTime".
        Error: Unknown type: "AWSDateTime".
```

Am I missing something here? Also, while on the topic, I can see that there are many places where `config/scalars` can be defined i.e in various plugins. Is it possible to define it globally somehow?

**Comment by dotansimha** [comment, marked as answer]
https://github.com/dotansimha/graphql-code-generator/discussions/4311
